### PR TITLE
Clone Images for cloned page

### PR DESCRIPTION
**Pull request by dasdib**
## Changes

- Copy images for cloned page

## Related Issues

- [Duplicating Images on Cloned paged #1177](https://github.com/johnsonandjohnson/Bodiless-JS/issues/1177)

### CODE REVIEW

**Review by konst8 — CHANGES_REQUESTED** [review]

File: packages/bodiless-backend/src/page.js [hunk 1]
```diff
@@ -235,20 +238,110 @@ class Page {
...
+    const originPagePath = `${basePath}${originPath}`;
+    const destinationPagePath = `${basePath}${destinationPath}`;
+
+    try {
+      // Check if any directory exist at origin file path and it has any files.
+      if (fs.existsSync(originStaticPath)) {
+        // Scan origin directory (page being cloned) to get list of existing directories name
+        const subDirsAtOrigin = await Page.dirHasDirectories(
+          originPagePath,
+        ).then((dirent) => dirent.map((dirent) => dirent.name));
```
[the remark this answers is not in the document]
Comment (dasdib) [reply packages/bodiless-backend/src/page.js:277]: Actually fs.readdir() returns dirent . https://nodejs.org/api/fs.html#fsreaddirpath-options-callback

File: packages/bodiless-backend/src/backend.js [hunk 2]
```diff
@@ -104,32 +105,24 @@ class GitCommit {
...
-      .exec();
+    const dirty = await GitCmd.cmd().add('diff', '--quiet').exec();
     if (dirty.code) {
-      await GitCmd.cmd()
-        .add('add', '--all')
-        .exec();
-      await GitCmd.cmd()
-        .add('commit', '-m', 'TEMPORARY COMMIT')
-        .exec();
+      await GitCmd.cmd().add('add', '--all').exec();
```
[the remark this answers is not in the document]
Comment (dasdib) [reply packages/bodiless-backend/src/backend.js]: @konst8 I have reverted it back as changed code moved to page.js only.

File: packages/bodiless-backend/src/backend.js [hunk 3]
```diff
@@ -31,10 +31,13 @@ const backendPrefix = process.env.GATSBY_BACKEND_PREFIX || '/___backend';
...
-const backendPagePath = process.env.BODILESS_BACKEND_DATA_PAGE_PATH || defaultBackendPagePath;
+const backendPagePath =
+  process.env.BODILESS_BACKEND_DATA_PAGE_PATH || defaultBackendPagePath;
 const backendStaticPath = process.env.BODILESS_BACKEND_STATIC_PATH || '';
-const backendPublicPath = process.env.BODILESS_BACKEND_PUBLIC_PAGE_PATH || 'public/page-data';
-const isExtendedLogging = (process.env.BODILESS_BACKEND_EXTENDED_LOGGING_ENABLED || '0') === '1';
+const backendPublicPath =
+  process.env.BODILESS_BACKEND_PUBLIC_PAGE_PATH || 'public/page-data';
+const isExtendedLogging =
+  (process.env.BODILESS_BACKEND_EXTENDED_LOGGING_ENABLED || '0') === '1';
```
[the remark this answers is not in the document]
Comment (dasdib) [reply packages/bodiless-backend/src/backend.js]: @konst8 I have moved the code changes to page.js only. Above section of code are untouched.

File: packages/bodiless-backend/src/page.js [hunk 4]
```diff
@@ -235,20 +238,110 @@ class Page {
...
 
+  static clonePageImgAssets(origin, destination, basePath) {
+    Page.clonePageAssets(origin, destination, basePath, '/images/pages');
+  }
+
+  static async clonePageAssets(origin, destination, basePath, target) {
+    const originPath = origin.replace(/\/$/, '');
+    const destinationPath = destination.replace(/\/$/, '');
+
+    const originStaticPath = `${backendStaticPath}${target}${originPath}`;
```
[the remark this answers is not in the document]
Comment (dasdib) [reply packages/bodiless-backend/src/page.js]: @konst8  Updated with path.join() with new push.

**Review by konst8 — APPROVED** [review]

**Review by irkhodzka — APPROVED** [review]

**Pull request merged**